e2e/exec.go

```diff
@@ -172,15 +171,37 @@ func unprivilegedCommandArray() []string {
 	}
 }
 
-func logSSHInstructions(s *Scenario) {
-	result := "SSH Instructions:"
+func logSSHInstructions(ctx context.Context, s *Scenario) {
+	privateIP, err := getVMPrivateIPAddress(ctx, s)
+	if err != nil {
+		s.T.Logf("error getting VM private IP: %v", err)
+		return
+	}
+	err = uploadSSHKey(ctx, s.Runtime.Cluster.Kube, privateIP, s.Runtime.Cluster.DebugPod.Name, s.Runtime.SSHKeyPrivate)
+	if err != nil {
+		s.T.Logf("error uploading SSH key to VM: %v", err)
+		return
+	}
+	result := "SSH Instructions. It may take some time for the VM to be ready for SSH."
 	if !config.Config.KeepVMSS {
 		result += " (VM will be automatically deleted after the test finishes, set KEEP_VMSS=true to preserve it or pause the test with a breakpoint before the test finishes)"
 	}
 	result += "\n========================\n"
 	result += fmt.Sprintf("az account set --subscription %s\n", config.Config.SubscriptionID)
 	result += fmt.Sprintf("az aks get-credentials --resource-group %s --name %s --overwrite-existing\n", config.ResourceGroupName, *s.Runtime.Cluster.Model.Name)
-	result += fmt.Sprintf(`kubectl exec -it %s -- bash -c "chroot /proc/1/root /bin/bash -c '%s'"`, s.Runtime.DebugHostPod, sshString(s.Runtime.VMPrivateIP))
+	result += fmt.Sprintf(`kubectl exec -it %s -- bash -c "chroot /proc/1/root /bin/bash -c '%s'"`, s.Runtime.Cluster.DebugPod.Name, sshString(s.Runtime.VMPrivateIP))
 	s.T.Log(result)
 	//runtime.Breakpoint() // uncomment to pause the test
 }
+
+func uploadSSHKey(ctx context.Context, kube *Kubeclient, vmPrivateIP, jumpboxPodName string, sshPrivateKey []byte) error {
+	cmd := fmt.Sprintf(`echo '%s' > %[2]s && chmod 0600 %[2]s`, sshPrivateKey, sshKeyName(vmPrivateIP), sshString(vmPrivateIP))
```

Comment:
This probably won't work on Windows.

Reply:
it's extracted from `execScriptOnVm`

Reply:
going to revert it, I've written before your changes.

Will add later once I encounter the issue again